pkg/apis/core/validation/validation.go

```diff
@@ -5332,3 +5333,32 @@ func ValidateProcMountType(fldPath *field.Path, procMountType core.ProcMountType
 		return field.NotSupported(fldPath, procMountType, []string{string(core.DefaultProcMount), string(core.UnmaskedProcMount)})
 	}
 }
+
+// validateTopologySpreadConstraints validates given TopologySpreadConstraints.
+func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	supportedRespModes := sets.NewString(string(core.DoNotSchedule), string(core.ScheduleAnyway))
+	// tuple {topologyKey, whenUnsatisfiable} denotes one kind of constraint
+	allConstraintKinds := sets.String{}
+	for i, constraint := range constraints {
+		subFldPath := fldPath.Index(i)
+		if val := constraint.MaxSkew; val <= 0 {
+			allErrs = append(allErrs, field.Invalid(subFldPath.Child("maxSkew"), val, isNotPositiveErrorMsg))
+		}
+		if len(constraint.TopologyKey) == 0 {
+			allErrs = append(allErrs, field.Required(subFldPath.Child("topologyKey"), "can not be empty"))
+		}
+		if !supportedRespModes.Has(string(constraint.WhenUnsatisfiable)) {
+			allErrs = append(allErrs, field.NotSupported(subFldPath.Child("whenUnsatisfiable"), constraint.WhenUnsatisfiable, supportedRespModes.List()))
+		}
+		kind := fmt.Sprintf("{%s, %s}", constraint.TopologyKey, constraint.WhenUnsatisfiable)
```

Comment:
it's slightly safer to use a struct{string,string} as a key.

Reply:
(imagine topology keys with commas in them)

Reply:
done.